src/providers/address/address.ts

```diff
@@ -20,9 +20,8 @@ export class AddressProvider {
     return extractedAddress;
   }
 
-  public getCoinAndNetwork(str: string): any {
+  public getCoinAndNetwork(str: string, network: string = 'livenet'): any {
     const address = this.extractAddress(str);
-    let network = null;
     try {
       network = this.bitcore.Address(address).network.name;
```

Comment:
I am not convinced on adding the `network` parameter to getCoinAndNetwork, but can't think a better way now.
 lets:
 - set `network = null` be default in params
 - add a `throw` if `network` param is given and  is != address's network for BTC/BCH, to avoid confusion